feat(nextcloud): #DRIV-26 add new route for copying and moving files from nextcloud

aymericdevriesere opened the pull request:
feat(nextcloud): #DRIV-26 add new route for copying and moving files from nextcloud

aymericdevriesere commented on src/main/java/fr/openent/nextcloud/service/impl/DefaultDocumentsService.java:
```diff
@@ -295,15 +302,286 @@ public Future<JsonArray> uploadFiles(UserNextcloud.TokenProvider userSession, Li
+                            });
+                        }
+                        else {
+                            //TODO Gérer le cas d'import d'un dossier
```
Les deux TODO dans les handlers sont pour une fonctionnalité à déterminer avec Sarah qui est en vacances !

qpic-cgi replied:
okey, du coup mais ta PR en DRAFT ou en WIP pour le moment

qpic-cgi commented on src/main/java/fr/openent/nextcloud/service/impl/DefaultDocumentsService.java:
```diff
@@ -295,15 +302,286 @@ public Future<JsonArray> uploadFiles(UserNextcloud.TokenProvider userSession, Li
+     * @param parentId          Id of the previous folder if moving in a folder
+     * @return                  Future Json with the infos about every move
+     */
+    private java.util.function.Function<JsonObject, Future<JsonObject>> moveHandler(UserNextcloud.TokenProvider userSession,
```
?
ça retourne pas directement Future<JsonObject> plutôt ?

qpic-cgi commented on src/main/java/fr/openent/nextcloud/service/impl/DefaultDocumentsService.java:
```diff
@@ -295,15 +302,286 @@ public Future<JsonArray> uploadFiles(UserNextcloud.TokenProvider userSession, Li
+                                                                             String file,
+                                                                             JsonObject result,
+                                                                             String parentId) {
+        return v -> {
```
créer une promesse et retourne là plutôt

aymericdevriesere commented:
## Describe your changes
Adding two new routes to API. The first one will be used to move a file from Nextcloud to workspace and the second one will be used to copy a file from Nextcloud to workspace. Nextcloud and Workspace need to belong to the same user.
## Checklist tests
Tested with postman
## Issue ticket number and link
[ DRIV-26 ]
https://entsupport.gdapublic.fr/projects/DRIV/issues/DRIV-26
## Checklist before requesting a review (magic string, indentation, comment/documentation...)

- [ ] I have detailed the tests to do in my feature/fix in order to prevent consequences regressions (must specify in **Checklist tests**)
- [ ] I have performed a self-review of my code
- [ ] I have commented my code, particularly in hard-to-understand areas
- [ ] I have made corresponding changes to the documentation (API Doc etc...) - (must specify in **Description** for target version)
- [ ] If it is a consequence feature, I have added thorough tests.
- [ ] I have added tests that prove my fix is effective or that my feature works
- [ ] Any dependent changes have been added to this project (must specify in **Description**)
